.github/workflows/build.yml

```diff
@@ -5,45 +5,84 @@ env:
 jobs:
   build:
     runs-on: ubuntu-22.04
+    permissions:
+      packages: write
+      contents: read
     steps:
       - name: Checkout Source Code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+
+      - name: Download linux
+        run: make download
 
       - name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v2
 
       - name: Set up Docker Buildx
         id: buildx
-        uses: docker/setup-buildx-action@v1
+        uses: docker/setup-buildx-action@v2
 
       - name: Docker meta
         id: docker_meta
-        uses: docker/metadata-action@v3
+        uses: docker/metadata-action@v4
+        with:
+          images: docker.io/${{ github.repository_owner }}/toolchain
+          tags: |
+            type=ref,event=branch
+            type=semver,pattern={{version}}
+
+      - name: GitHub Registry meta
+        id: ghcr_meta
```

Comment:
Not there yet 🙂
Use a single meta
Just add a condition like [this](https://github.com/cartesi/rollups/blob/40aa6af3ab7668c698e9b3eac03621b3c585c11f/.github/workflows/docker.yml#L39)